dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dataaccess/FileAccessIO.java

```diff
@@ -574,7 +576,7 @@ private String getDatasetDirectory() throws IOException {
 
             baseName = this.getDataFile().getStorageIdentifier();
 
-            datasetDirectoryPath = this.getDataFile().getOwner().getFileSystemDirectory();
+            datasetDirectoryPath = this.getDataFile().getOwner().getFileSystemDirectory(System.getProperty("dataverse.files.directory"));
```

Comment:
Instead of `System.getProperty("dataverse.files.directory")` use additional parameter that will be set in constructor.
And modify DataAccess accordingly